### PR TITLE
Enhance ArrayCopy for 16-bit primitives on X86

### PR DESCRIPTION
**Pull request by 0dvictor**
Use REP MOVSD to copy 16-bit primitives instead of REP MOVSW.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: compiler/x/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -3992,6 +4049,7 @@ TR::Register *OMR::X86::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::Co
 
    TR::DataType dt = node->getArrayCopyElementType();
    uint32_t elementSize = 1;
+   static bool avoidREPMOVSW = feGetEnv("TR_AvoidREPMOVSW");
```
Comment (0xdaryl) [inline compiler/x/codegen/OMRTreeEvaluator.cpp]: Please provide a slightly more descriptive env var to indicate context.  e.g., `TR_AvoidREPMOVSWForArrayCopy`

File: compiler/x/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -3568,6 +3568,63 @@ static void arrayCopy64BitPrimitiveOnIA32(TR::Node* node, TR::Register* dstReg,
    cg->stopUsingRegister(scratch);
    }
```
[the remark this answers is not in the document]
Comment (0dvictor) [reply compiler/x/codegen/OMRTreeEvaluator.cpp]: Good suggestion, updated.

File: compiler/x/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (0dvictor) [reply compiler/x/codegen/OMRTreeEvaluator.cpp]: Sure, it has been updated.

**Comment by 0xdaryl** [comment]
Genie build is not required as x86-only files were modified.

**Pull request merged**